libs/ktem/ktem/reasoning/react.py

```diff
@@ -0,0 +1,322 @@
+import html
+import logging
+from typing import AnyStr, Optional, Type
+
+from ktem.llms.manager import llms
+from ktem.reasoning.base import BaseReasoning
+from ktem.utils.render import Render
+from langchain.text_splitter import CharacterTextSplitter
+from pydantic import BaseModel, Field
+
+from kotaemon.agents import (
+    BaseTool,
+    GoogleSearchTool,
+    LLMTool,
+    ReactAgent,
+    WikipediaTool,
+)
+from kotaemon.base import BaseComponent, Document, HumanMessage, Node, SystemMessage
+from kotaemon.llms import ChatLLM, PromptTemplate
+
+logger = logging.getLogger(__name__)
+
+
+class DocSearchArgs(BaseModel):
+    query: str = Field(..., description="a search query as input to the doc search")
+
+
+class DocSearchTool(BaseTool):
```

Comment:
Should DocSearchTool belong to kotaemon.agents.tools?

Reply:
Should move to here: https://github.com/Cinnamon/kotaemon/issues/60